tests/JsonTest.php

```diff
@@ -29,6 +31,7 @@ public function testIsHandlingInvalidDataForEncode()
 
     /**
      * @expectedException \LogicException
+     * @expectedExceptionMessage Invalid JSON: Malformed UTF-8 characters, possibly incorrectly encoded
```

Comment:
If we are refactoring this, why not use `$this->expectException()` and such instead? It makes the test much better to read in a lot of cases and is what we use in our other applications.

Reply:
We're not refactoring, just adding a new feature.

Reply:
You're right, it is just adding the message. However, this is a good opportunity to do a small refactoring. But I'll leave it be either way.